src/main/java/edu/harvard/iq/dataverse/EditDatafilesPage.java

```diff
@@ -2562,20 +2562,6 @@ public boolean isLocked() {
         return false;
     }
     
-    public boolean isThumbnailAvailable(FileMetadata fileMetadata) {
-        // new and optimized logic: 
-        // - check download permission here (should be cached - so it's free!)
-        // - only then ask the file service if the thumbnail is available/exists.
-        // the service itself no longer checks download permissions.  
-        if (!fileDownloadHelper.canDownloadFile(fileMetadata)) {
-            return false;
-        }
-
-        return datafileService.isThumbnailAvailable(fileMetadata.getDataFile());
-    }
```

Comment:
Here is the canDownloadFile logic that was removed, as I do no believe is necessary on EditDatafilesPage. If you can edit, you can download.

Reply:
While this is true with the current built in roles, it is not inherently true of the permission system. Is it possible that some day, someone would want a a role that can edit but not download (for example editing the metadata?) It may be worth keeping this; alternatively we can keep removed knowing that we'd have to revisit it someday.

Reply:
note: added the permission check back (directly in the rendered of the fragment).